sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,180 @@
+#include "opentelemetry/sdk/trace/tracer.h"
+#include "opentelemetry/sdk/trace/sampler.h"
+#include "opentelemetry/sdk/trace/samplers/always_off.h"
+#include "opentelemetry/sdk/trace/samplers/always_on.h"
+#include "opentelemetry/sdk/trace/samplers/parent_or_else.h"
+#include "opentelemetry/sdk/trace/samplers/probability.h"
+#include "opentelemetry/sdk/trace/simple_processor.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include <cstdint>
+
+#include <benchmark/benchmark.h>
+
+using namespace opentelemetry::sdk::trace;
+namespace nostd  = opentelemetry::nostd;
+namespace common = opentelemetry::common;
+using opentelemetry::trace::SpanContext;
+
+/**
+ * A mock exporter that switches a flag once a valid recordable was received.
+ */
+class MockSpanExporter final : public SpanExporter
```

Comment:
It would make sense to factor this out into a `mock_span_exporter.h`, or a `inmemory_span_exporter.h`. Now we have the code duplicated in `tracer_test.cc` and `sampler_benchmark.cc`.

Reply:
I was planning on doing that in a different PR right after this was merged. Would you rather I address that here?